multicluster/controllers/multicluster/member/gateway_controller.go

```diff
@@ -185,19 +203,56 @@ func (r *GatewayReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
-func (r *GatewayReconciler) getClusterInfo(gateway *mcsv1alpha1.Gateway) *mcsv1alpha1.ClusterInfo {
-	clusterInfo := &mcsv1alpha1.ClusterInfo{
+func (r *GatewayReconciler) clusterSetMapFunc(a client.Object) []reconcile.Request {
+	clusterSet := &mcv1alpha2.ClusterSet{}
+	requests := []reconcile.Request{}
+	if a.GetNamespace() != r.namespace {
+		return requests
+	}
+	ctx := context.TODO()
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: a.GetNamespace(), Name: a.GetName()}, clusterSet)
+	if err == nil {
+		if len(clusterSet.Status.Conditions) > 0 && clusterSet.Status.Conditions[0].Status == v1.ConditionTrue {
+			r.setClusterSetReady(true)
+			gwList := &mcv1alpha1.GatewayList{}
+			r.Client.List(ctx, gwList, &client.ListOptions{Namespace: r.namespace})
+			requests = make([]reconcile.Request, len(gwList.Items))
+			for i, gw := range gwList.Items {
+				requests[i] = reconcile.Request{
+					NamespacedName: types.NamespacedName{
+						Namespace: gw.Namespace,
+						Name:      gw.Name,
+					},
+				}
+			}
+		}
+	} else if apierrors.IsNotFound(err) {
+		r.setClusterSetReady(false)
```

Comment:
I do not think we should set the flag in a map func. First, let us see if we can just use `GetRemoteCommonAreaAndLocalID` . If really not, we can let ClusterSet controller set the flag (and we just need a single shared flag in ClusterSet controller).